### ISSUE TITLE
Extension lists in errors can break the API specification

### ISSUE THREAD

**Issue by bushjames** [mojaloop/central-services-error-handling#113]
Occasionally we have seen errors (returned from ALS) with extension lists that are not API spec compliant due to the "value" properties of an extension being longer than the maximum stated 128 characters (API spec document section 7.3.14). I suggest this library should check for, and possibly truncate invalid strings in both key and value properties.

**Comment by elnyry** [comment]
@bushjames just posting a quick response here..

This 'truncation' part is actually configurable and this is being done at the adapter level, so in ALS it can be set to 'true' and the length will be truncated to 128 to conform to the Spec. For example, in ml-api-adapter we see this set to 'true', whereas in central-ledger its set to 'false'.

The thinking at the time was that the information such as debug info, traces, etc can be used within internal services as needed but these will be truncated when its send outside of the Switch. For example [here](https://github.com/mojaloop/ml-api-adapter/blob/f2af1b7bc3c4575d1fadcf2656c0520db43a91bf/config/default.json#L20) is the config in ml-api-adapter that sets this to _**false**_

Please let me know if this is good enough or if you still think additional work is needed

**Comment by bushjames** [comment]
Thanks for the fast feedback @elnyry. As we are seeing this on ALS errors, would you be comfortable with me changing the default config on ALS to be the same as the ml-api-adapter?

**Comment by elnyry** [comment]
of course, it should be 'true' here as messages are sent out from it (same should apply for quoting-service)

**Comment by bushjames** [comment]
Many thanks @elnyry . I will get it sorted.

**Comment by elnyry** [comment]
Thanks @bushjames . Just checked and it's set to 'true' here, so the extension values should be truncated, unless this config is overriden - https://github.com/mojaloop/account-lookup-service/blob/1244e71f572aad8faf5e91df2baf9efd0ad631f2/config/default.json#L28 ..

However, if you're still seeing this then it could be a bug.

**Comment by bushjames** [comment]
I will investigate further.
